### PR TITLE
Option parser: fix warning, rename function and shrink code

### PR DESCRIPTION
**Pull request by eksperimental**
Fixes warning introduced in https://github.com/elixir-lang/elixir/pull/5709
/cc @AndrewDryga

### CODE REVIEW

**Comment by eksperimental** [comment]
Reason for renaming do_next to next_tagged is that do_ is usually used for a private version of a function, but here next is already private, so it's a non-descriptive. Besides I think we want to get rid of the whole do_ prefix

File: lib/elixir/lib/option_parser.ex [hunk 1]
```diff
@@ -712,6 +707,7 @@ defmodule OptionParser do
 
   defp to_existing_key(option, true),
     do: String.to_atom(option)
+
```
Comment (josevalim) [inline lib/elixir/lib/option_parser.ex]: Please keep them together since the one above is using single line definitions.

**Comment by josevalim** [comment]
Thank you! I have merged the first commit manually. We will skip the second one to avoid style changes.  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

Also, we are moving away from `do_` prefixes altogether, so good call!

File: lib/elixir/lib/option_parser.ex (hunk 1 above)
Comment (eksperimental) [inline lib/elixir/lib/option_parser.ex]: yes, but i thought we keep them together when the following clauses using the same two line definition,

**Pull request closed without merging**